test/test_scorep.py

```diff
@@ -375,6 +375,7 @@ def test_threads(scorep_env, instrumenter):
 @foreach_instrumenter
 def test_io(scorep_env, instrumenter):
     trace_path = get_trace_path(scorep_env)
+    scorep_env["SCOREP_IO_POSIX"] = "true"
```

Comment:
just an idea: maybe set this env also if `--io=posix` was specified on the command line, that smoothes the transition for Python users. `scorep(-config)` prints a decprecation warning, the bindings should print, that they exported `SCOREP_IO_POSIX=true`, the latter only if it is not yet set in the env